larsim/MCSTReco/MCReco_module.cc

```diff
@@ -95,28 +109,32 @@ void MCReco::produce(art::Event & evt)
   }
 
   const std::vector<simb::MCParticle>& mcp_array(*mcpHandle);
-  fPart.AddParticles(mcp_array,orig_array);
 
+  if (fIncludeDroppedParticles) {
+    art::Handle<std::vector<sim::MCMiniPart> > mcmpHandle;
+    evt.getByLabel(fMCMiniPartLabel,mcmpHandle);
+    if(!mcmpHandle.isValid()) throw cet::exception(__FUNCTION__) << "Failed to retrieve sim::MCMiniPart";;
+
+    const std::vector<sim::MCMiniPart>& mcmp_array(*mcmpHandle);
+
+    fPart.AddParticles(mcp_array, orig_array, mcmp_array);
+
+  } // end if fIncludeDroppedParticles
+  else {
+    fPart.AddParticles(mcp_array,orig_array);
+  }
 
   // change implemented by David Caratelli to allow for MCRECO to run without SimChannels and using
   // SimEnergyDeposits instead
   if (fUseSimEnergyDeposit == true) {
-    // Retrieve SimEnergyDeposit
-    art::Handle<std::vector<sim::SimEnergyDeposit> > sedHandle;
-    evt.getByLabel(fSimChannelLabel,sedHandle);
-    if(!sedHandle.isValid()) throw cet::exception(__FUNCTION__) << "Failed to retrieve sim::SimEnergyDeposit";
-
-    const std::vector<sim::SimEnergyDeposit>&  sed_array(*sedHandle);
-    fEdep.MakeMCEdep(sed_array);
+    MakeMCEdep<sim::SimEnergyDeposit>(evt, "sim::SimEnergyDeposit");
```

Comment:
I think using the template in this case is probably overkill.  Just:

```c++
auto const& seds = *evt.getValidHandle<std::vector<sim::SimEnergyDeposit>>(fSimChannelLabel);
fEdep.MakeMCEdep(seds);
```
...and similar elsewhere.  If you decide to keep the template, it can be significantly shortened based on using `getValidHandle`.

Reply:
I kept the template but shortened it using `getValidHandle`.